processor/src/aux_table_bus/aux_trace.rs

```diff
@@ -0,0 +1,76 @@
+// ================================================================================================
+
+/// Describes how to construct execution traces of auxiliary trace columns that depend on multiple
+/// co-processors in the Auxiliary Table (used in multiset checks).
+pub struct AuxTraceBuilder {
+    pub(super) lookup_hints: Vec<(usize, AuxTableLookup)>,
+    pub(super) request_rows: Vec<AuxTableLookupRow>,
+    pub(super) response_rows: Vec<AuxTableLookupRow>,
+}
+
+impl AuxTraceBuilder {
+    // COLUMN TRACE CONSTRUCTOR
+    // --------------------------------------------------------------------------------------------
+
+    /// Builds and returns the Auxiliary Table's auxiliary trace columns. Currently this consists of
+    /// a single bus column `b_aux` describing co-processor lookups requested by the stack and
+    /// provided by the co-processors in the Auxiliary Table.
+    pub fn build_aux_columns<E: FieldElement<BaseField = Felt>>(
+        &self,
+        main_trace: &Matrix<Felt>,
+        rand_elements: &[E],
+    ) -> Vec<Vec<E>> {
+        let b_aux = self.build_aux_column(main_trace, rand_elements);
+        vec![b_aux]
+    }
+}
+
+// AUXILIARY TABLE LOOKUPS
+// ================================================================================================
+
+impl AuxColumnBuilder<AuxTableLookup, AuxTableLookupRow> for AuxTraceBuilder {
+    /// Returns a list of lookup rows which were requested by stack operations.
+    ///
+    /// The order of the rows in the list is the same as the order in which the requests were sent.
+    fn get_table_rows(&self) -> &[AuxTableLookupRow] {
+        &self.request_rows
+    }
+
+    /// Returns hints which describe the [AuxTable] lookup requests and responses during program
+    /// execution. Each update hint is accompanied by a clock cycle at which the update happened.
+    ///
+    /// Internally, each update hint also contains an index of the row into the full list of request
+    /// rows or response rows, depending on whether it is a request, a response, or both (in which
+    /// case it contains 2 indices).
+    fn get_table_hints(&self) -> &[(usize, AuxTableLookup)] {
+        &self.lookup_hints
+    }
+
+    /// Returns the value by which the running product column should be multiplied for the provided
+    /// hint value.
+    fn get_multiplicand<E: FieldElement<BaseField = Felt>>(
+        &self,
+        hint: AuxTableLookup,
+        _row_values: &[E],
+        inv_row_values: &[E],
+        alphas: &[E],
+    ) -> E {
+        match hint {
+            AuxTableLookup::Request(request_row) => inv_row_values[request_row],
+            AuxTableLookup::Response(response_row) => {
+                self.response_rows[response_row].to_value(alphas)
+            }
+            AuxTableLookup::RequestAndResponse((request_row, response_row)) => {
+                inv_row_values[request_row] * self.response_rows[response_row].to_value(alphas)
+            }
+        }
```

Comment:
You might be able to override `build_row_values()` (and leave `get_table_rows()` unimplemented) to build both row row and inverse row values there. This way, you won't need to pass `alphas` here.